src/tests/ftest/util/AgentUtils.py

```diff
@@ -102,20 +126,71 @@ def run_agent(basepath, client_list, server_list):
                              stderr=subprocess.PIPE)
         sessions[client] = p
 
-
-def stop_agent(client_list):
+    # double check agent launched successfully
+    file_desc = sessions[client].stderr.fileno()
+    flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
+    fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
+    timeout = 5
+    start_time = time.time()
+    result = 0
+    pattern = "Starting daos_agent"
+    expected_data = ""
+    while True:
+        output = ""
+        try:
+            output = sessions[client].stderr.read()
+        except IOError as excpn:
+            if excpn.errno != errno.EAGAIN:
+                raise excpn
+            continue
+        match = re.findall(pattern, output)
+        expected_data += output
+        result += len(match)
+        if not output or time.time() - start_time > timeout:
+            print("<AGENT>: {}".format(expected_data))
+            raise AgentFailed("DAOS Agent didn't start!")
+        break
+    print("<AGENT> agent started and took %s seconds to start" % \
+            (time.time() - start_time))
+
+    return sessions
+# pylint: enable=too-many-locals
+
+
+def stop_agent(client_list, sessions):
     """
     This should kill ssh and the agent.  This is temporary; presuming the
     agent will deamonize at somepoint and can be started killed more
     appropriately.
 
     client_list -- kill daos_agent on these hosts
+    sessions    -- set of subprocess sessions returned by run_agent()
```

Comment:
Really should just be `session` since it's not a global of all sessions, but just represents a single session, which the caller tracks:
```suggestion
    session    -- set of subprocess sessions returned by run_agent()
```